docs/extending-machetli.rst

```diff
@@ -128,19 +128,17 @@ generators can directly access entities like sections, included packages, etc.
     :linenos:
 
     def generate_initial_state(filename):
-        with open(filename) as f:
-            content = f.read()
+        content = Path(filename).read_text()
```

Comment:
Even though this is also a user-facing function and only an example, I'd still change this to
```Python
def generate_initial_state(path: Path):
        content = path.read_text()
```